cdap-docs/examples-manual/source/examples/log-analysis.rst

```diff
@@ -0,0 +1,205 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :description: Cask Data Application Platform SparkPageRank Application
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _examples-spark-page-rank:
+
+====================
+Log Analysis Example
+====================
+
+A Cask Data Application Platform (CDAP) example demonstrating Spark and MapReduce
+running in parallel inside a Workflow through fork.
+
+Overview
+========
+
+This example demonstrates Spark and MapReduce performing log analysis, computing
+total number of hits for every unique URL, total number of responses for every
+unique response code, and total number of requests made by every unique IP address,
+based on Apache usage log.
+
+Logs are sent to CDAP and ingested into the *logStream*, which stores the log
+information event in its entirety.
+
+After these events are streamed, they are taken up by the *ResponseCounterSpark*, which
+goes through the entries, calculates the total number of responses for every unique
+response code, and tabulates results in an ``KeyValueTable`` dataset, *responseCount*.
+The Spark program also computes the total number of requests made by every unique IP
+address and writes it to ``TimePartitionedFileSet``, *reqCount*.
+
+In parallel, these events are also taken up by the *HitCounterProgram*, which goes
+through the entries, calculates the total number hits for every unique URL and
+tabulates results in a ``KeyValueTable`` dataset, *hitCount*.
+
+The *LogAnalysisWorkflow* ties the Spark and MapReduce programs to run in parallel.
+
+Once the application completes, you can query the *responseCount* dataset by using
+the ``rescount`` endpoint of the *ResponseCounterService*. It will send back a
+string result with the total number of responses on the ``rescount`` query parameter.
+You can also query the *hitCount* dataset by using the ``url`` endpoint of the
+*HitCounterService*. It will send the total number of hits for the queried url.
+You can query the ``reqCount`` ``TimePartitionedFileSet`` by using the ``reqcount``
+endpoint of the *RequestCounterService* which will return a set of all available
+partitions. Using one of partitions from the above set, you can query for the total
+number of requests made by every unique IP address in last 60 minutes. The ``reqfile``
+endpoint of the *RequestCounterService* returns a map of IP addresses to the total
+number of requests made by them.
+
+Let's look at some of these components, and then run the application and see the results.
+
+The LogAnalysis Application
+---------------------------
+
+As in the other `examples <index.html>`__, the components
+of the application are tied together by the class ``LogAnalysisApp``:
+
+.. literalinclude:: /../../../cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java
+   :language: java
+   :lines: 70-94
+
+The *hitCount* and  *responseCount* ``KeyValueTable``s, and the  *reqCount* ``TimePartitionedFileSet``
```

Comment:
This is not formatting correctly, because of the lack of a space at the end of the ``KeyValueTable``. But we don't want a gap, so use:
```
*hitCount* and *responseCount* ``KeyValueTables`` and *reqCount* ``TimePartitionedFileSet``